### PR TITLE
Changes to facilitate pulling the KFS Account information out of the Xcomment field.

### PR DESCRIPTION
**Pull request by taylorkj**
@srkirkland ; @jpknoll ; @jSylvestre:
This will work.  Note that there will ALWAYS be a result.Groups array consisting of 4 elements because of the way the regex was defined; therefore, it is not necessary to do an Any() check; however, the length of the elements at pos'n 1 and 2 need to be checked to verify they are not blank.

### CODE REVIEW

File: GiftModels/PremiumDetails.cs [hunk 1]
```diff
@@ -52,22 +49,66 @@ public string PremiumChart
...
         /// This field is defaulted from the tms_premium table (W3). It is the monetary value associated with this premium. This value is normally the same amount as the value in the tms_premium table. 
         /// </summary>
         public decimal PremiumAmount { get; set; }
+
+        /// <summary>
+        /// Parse out the KFS chart, Account, and Sub-account (if present) from the XComment field. 
+        /// </summary>
+        private void InitAccounDetails()
+        {
+            const string sPattern = @"\[\[(\w)-(\w{5,7})-?(\w{5})?\]\]";
```
Comment (jpknoll) [inline GiftModels/PremiumDetails.cs]: :-D

File: GiftModels/PremiumDetails.cs [hunk 2]
```diff
@@ -41,8 +37,9 @@ public string PremiumChart
             {
                 if (string.IsNullOrWhiteSpace(_premiumChart) && string.IsNullOrWhiteSpace(Xcomment))
                     return null;
-                if (string.IsNullOrWhiteSpace(_premiumChart) && Xcomment.Contains("-"))
-                    return Xcomment.Substring(0, Xcomment.IndexOf('-'));
+                if (string.IsNullOrWhiteSpace(_premiumChart))
+                    InitAccounDetails();
```
Comment (jSylvestre) [inline GiftModels/PremiumDetails.cs]: @taylorkj Minor, but method name is missing a t

File: GiftModels/PremiumDetails.cs [hunk 3]
```diff
@@ -52,22 +49,66 @@ public string PremiumChart
...
+
+        /// <summary>
+        /// Parse out the KFS chart, Account, and Sub-account (if present) from the XComment field. 
+        /// </summary>
+        private void InitAccounDetails()
+        {
+            const string sPattern = @"\[\[(\w)-(\w{5,7})-?(\w{5})?\]\]";
+            var result = System.Text.RegularExpressions.Regex.Match(Xcomment, sPattern);
+
+            if (result.Groups[1].Length > 0 && result.Groups[1].ToString().Length > 0)
```
Comment (jpknoll) [inline GiftModels/PremiumDetails.cs]: @taylorkj Not exactly sure what this if statement is checking for. You might check that Group[0].ToString() isn't empty.

Comment (taylorkj) [inline GiftModels/PremiumDetails.cs]: @jpknoll : My intention was to make sure that we could pull non-empty strings for chart and account.  Checking just the Groups[0] doesn't necessarily tell me that, I think.

File: GiftModels/PremiumDetails.cs [hunk 4]
```diff
@@ -41,8 +37,9 @@ public string PremiumChart
             {
                 if (string.IsNullOrWhiteSpace(_premiumChart) && string.IsNullOrWhiteSpace(Xcomment))
                     return null;
-                if (string.IsNullOrWhiteSpace(_premiumChart) && Xcomment.Contains("-"))
-                    return Xcomment.Substring(0, Xcomment.IndexOf('-'));
+                if (string.IsNullOrWhiteSpace(_premiumChart))
+                    InitAccountDetails();
```
Comment (jSylvestre) [inline GiftModels/PremiumDetails.cs]: @taylorkj For clarity, I prefer to always wrap if statements with {} even if it is a single line. But I don't know if that is one of our preferences for our group.

**Pull request merged**